mapboxgl/templates/main.html

```diff
@@ -126,6 +126,41 @@
 }
 
 
+function formatJavascriptStack(stack) {
+    var lines = stack.split("\n");
+    var formattedStack = "";
+    for (var i = 0; i < lines.length; i++) {
+        var trace = lines[i].split("@");
+        if (trace.length > 1) {
+            trace[0] = trace[0].replace("\]\<\/", "\]").replace("\<\/", ":");
+            var lineTemplate = `<span class='ansi-green-intense-fg ansi-bold'>${trace[1]}</span>`
+            if (trace[0] != "") {
+                lineTemplate +=  ` in <span class='ansi-cyan-fg'>${trace[0]}</span>\n`;
+                formattedStack +=  lineTemplate;
+            }
+        }
+    }
+    return formattedStack;
+}
+
+
+function formatJavascriptError(errorName, errorObj) {
+    var formattedStack = formatJavascriptStack(errorObj.stack);
+    var formattedError = `<pre>
+<span class="ansi-red-intense-fg ansi-bold">---------------------------------------------------------------------------</span>
+<span class="ansi-red-intense-fg ansi-bold">${errorName}</span>                              Traceback (most recent call last)
+${formattedStack}
+<span class="ansi-red-intense-fg ansi-bold">${errorName}</span>: ${errorObj.message}</pre>`;
+    return formattedError;
+}
+
+window.onerror = function (errorMsg, url, lineNumber, column, errorObj) {
```

Comment:
this is a common need when the map has a valid style json, but the layer we add for the CircleViz, HeatmapViz, etc is not valid.

Perhaps display the error in a modal the user can dismiss?

Also, we probably only want to show some errors from the map here - not all errors.  For example, the map could throw a warning when requesting an invalid tile from a valid tileset.

Reply:
Alright sounds easy enough, I rebased this PR to get rid of the error handling commit so we can close it if you're fine with it.
I'll open a new PR after dinner with the error handling code and modal related modifications.

Reply:
Great!  Thanks @etienneDesticourt.  I think we're ready to merge pending @perrygeo 👍